lib/commands/administration.js

```diff
@@ -134,6 +134,99 @@ Cypress.Commands.add(
   }
 );
 
+Cypress.Commands.add('clearUserRoles', { prevSubject: 'optional' }, (...args) => {
+  const $args = normalizedArgumentsWithAuth(args);
+  const [auth, user, clientOptions] = $args;
+
+  const options = { ...clientOptions, ...{ failOnStatusCode: false } };
+  const consoleProps = {
+    auth: auth,
+    clientOptions: options,
+  };
+
+  Cypress.log({
+    name: 'clearUserRoles',
+    message: _.isObjectLike(user) ? user.userName : user,
+    consoleProps: () => consoleProps,
+  });
+
+  if (!user || (_.isObjectLike(user) && !user.userName)) {
+    return throwError(
+      'Missing argument. Requiring IUser object with userName or username argument.'
+    );
+  }
+
+  const userIdentifier = _.isObjectLike(user) ? user.userName : user;
+
+  return cy
+    .wrap(auth, { log: false })
+    .c8yclient((client) => client.user.detail(userIdentifier), options)
+    .then((response) => {
+      const assignedRoles = response.body.groups.references;
+
+      if (!assignedRoles || assignedRoles.length === 0) {
+        return cy.wrap(auth, { log: false });
+      }
+
+      cy.wrap(assignedRoles, { log: false }).each((assingedRole) => {
+        cy.wrap(auth, { log: false }).c8yclient(
+          (client) => client.userGroup.removeUserFromGroup(assingedRole.group.id, userIdentifier),
+          options
+        );
+      });
+
+      return cy.wrap(auth, { log: false });
+    });
+});
+
+Cypress.Commands.add('assignUserRoles', { prevSubject: 'optional' }, (...args) => {
+  const $args = normalizedArgumentsWithAuth(args);
+  const [auth, user, roles, clientOptions] = $args;
+
+  const options = { ...clientOptions, ...{ failOnStatusCode: false } };
+  const consoleProps = {
+    auth: auth,
+    clientOptions: options,
+  };
+
+  Cypress.log({
+    name: 'assignUserRoles',
+    message: _.isObjectLike(user) ? user.userName : user,
+    consoleProps: () => consoleProps,
+  });
+
+  if (!user || (_.isObjectLike(user) && !user.userName)) {
+    return throwError(
+      'Missing argument. Requiring IUser object with userName or username argument.'
+    );
+  }
+
+  if (!roles || roles.length === 0) {
+    return throwError('Missing argument. Requiring an string array with roles.');
+  }
+
+  const userIdentifier = _.isObjectLike(user) ? user.userName : user;
+
+  cy.wrap(roles, { log: false }).each((role) => {
+    cy.wrap(auth, { log: false }).c8yclient(
+      [
+        (client) => client.core.fetch(`/user/${client.core.tenant}/groupByName/${role}`),
+        (client, groupResponse) => {
+          const baseUrl = options && !!options.baseUrl ? options.baseUrl : client.core.baseUrl;
+
+          return client.userGroup.addUserToGroup(
+            groupResponse.body.id,
+            `https://${baseUrl}/user/${client.core.tenant}/users/${userIdentifier}`
```

Comment:
@CGuether Could be also an option to query the user object and than use `userResponse.body.self` instead of creating the link. Please also note that all requests have `failOnStatusCode` disabled, requiring error handling in case user or group does not exist.